pinot-core/src/main/java/com/linkedin/pinot/core/operator/docidsets/ScanBasedSingleValueDocIdSet.java

```diff
@@ -72,6 +74,16 @@ public long getNumEntriesScannedInFilter() {
     return blockValSetBlockDocIdIterator.getNumEntriesScanned();
   }
 
+  @Override
+  public long getNumIndicesLoaded() {
+    return 0L;
+  }
+
+  @Override
+  public long getTotalBytesRead() {
+    return getNumEntriesScannedInFilter() * dataTypeSizeInBytes;
```

Comment:
A better approximation of length for variable length data types would be avg length (right now it seems to use max length). This can be precomputed during segment generation and stored in the metadata.